api/src/main/java/edu/cornell/mannlib/vitro/webapp/i18n/VitroResourceBundle.java

```diff
@@ -127,8 +171,45 @@ private VitroResourceBundle(String bundleName, ServletContext ctx,
 	}
 
 	private Properties loadProperties() throws IOException {
+		Properties translations = new Properties();
+		translations = loadTranslationsFromTriplestore(translations);
+		translations = loadTranslationsFromFiles(translations);
+		return translations;
+	}
+
+	private Properties loadTranslationsFromTriplestore(Properties props) throws IOException {
+		ParameterizedSparqlString pss = new ParameterizedSparqlString();
+		pss.setCommandText(SPARQL_LANGUAGE_QUERY);
+		pss.setLiteral("current_theme", theme);
+		pss.setLiteral("current_application", applicationName);
+		if (displayModel != null){
+			Model queryModel = LanguageFilteringUtils.wrapOntModelInALanguageFilter(displayModel, locale);
+			try {
+				QueryExecution qexec = QueryExecutionFactory.create(pss.toString(), queryModel);
+				try {
+					ResultSet results = qexec.execSelect();
+
+					while (results.hasNext()) {
+						QuerySolution solution = results.nextSolution();
+						props.put(solution.get("key").toString(), solution.get("translation").toString());
+					}
+
+				} catch (Exception e) {
+					log.error(e.getLocalizedMessage());
+					e.printStackTrace();
```

Comment:
We normally use only the application-specific log and not e.printStackTrace(), which will pollute catalina.out.  Typically we log exceptions using the form `log.error(e, e)`, which will print the stack trace along with the message.  I don't know if it uses the localized message by default, but presumably by passing the whole exception to the logger we allow the logging configuration to determine the details of what's displayed rather than deciding in the code.